### ISSUE TITLE
Bug: no-unused-vars false positives when destructured in func sig and used via spread syntax

### ISSUE THREAD

**Issue by adamlui** [eslint/eslint#19390]
### Environment

Node version: v22.9.0
npm version: v10.9.1
Local ESLint version: v9.19.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
import js from '@eslint/js'
import globals from 'globals'
import css from '@eslint/css'
import html from '@html-eslint/eslint-plugin'
import htmlParser from '@html-eslint/parser'
import importPlugin from 'eslint-plugin-import'
import json from '@eslint/json'
import markdown from '@eslint/markdown'
import regexp from 'eslint-plugin-regexp'
import stylisticJS from '@stylistic/eslint-plugin-js'
import yml from 'eslint-plugin-yml'

export default [
    { ignores: ['**/*.min.js', '**/package-lock.json', 'docs/**/*.min.css', 'docs/**/footer.html'] },
    {
        files: ['**/*.{js,mjs}'],
        languageOptions: {
            ecmaVersion: 'latest', sourceType: 'script',
            globals: {
                ...globals.browser, ...globals.greasemonkey, ...globals.node, chatgpt: 'readonly', chrome: 'readonly',
                config: 'writable', dom: 'readonly', icons: 'writable', modals: 'writable', settings: 'writable'
            }
        },
        plugins: { 'import': importPlugin, 'js-styles': stylisticJS, regexp },
        rules: {
            ...js.configs.recommended.rules,
            ...importPlugin.flatConfigs.recommended.rules,
            ...regexp.configs['flat/recommended'].rules,
            'indent': 'off', 'no-unexpected-multiline': 'off', 'key-spacing': 'off', // allow whitespace anywhere
            'js-styles/no-trailing-spaces': 'error', // ...except at ends of lines
            'js-styles/max-len': ['error', { 'code': 120, // limit lines to 120 chars except if containing...
                'ignoreComments': true, 'ignoreStrings': true, // ...trailing/own-line comments, quoted strings...
                'ignoreTemplateLiterals': true, 'ignoreRegExpLiterals': true }], // ...or template/regex literals
            'js-styles/no-extra-semi': 'error', // disallow unnecessary semicolons
            'quotes': ['error', 'single', // enforce single quotes...
                { 'allowTemplateLiterals': true }], // ...except backticks to avoid escaping quotes
            'comma-dangle': ['error', 'never'], // enforce no trailing commas in arrays or objects
            'no-constant-condition': 'off', // allow constant conditions
            'no-empty': 'off', // allow empty blocks
            'no-inner-declarations': 'off', // allow function declarations anywhere
            'no-useless-escape': 'off', // allow all escape chars cause ESLint sucks at detecting truly useless ones
            'no-unused-vars': ['error', { 'caughtErrors': 'none' }] // allow unused named args in catch blocks
        }
    },
    { files: ['**/chatgpt.js'], languageOptions: { globals: { chatgpt: 'off' }}},
    { files: ['**/*.css'], language: 'css/css', ...css.configs.recommended },
    {
        files: ['**/*.html'], languageOptions: { parser: htmlParser }, plugins: { '@html-eslint': html },
        rules: {
            ...html.configs['flat/recommended'].rules,
            '@html-eslint/require-title': 'off', // allow missing title tags
            '@html-eslint/quotes': 'off', // allow unquoted attrs for compactness
            '@html-eslint/attrs-newline': 'off', // allow multi-attrs in single line
            '@html-eslint/element-newline': ['error', { 'skip': ['html'] }] // allow multi-tags in single line
        }
    },
    { files: ['**/*.json'], language: 'json/json', ...json.configs.recommended },
    {
        files: ['**/*.md'], language: 'markdown/commonmark', plugins: { markdown },
        rules: {
            ...markdown.configs.recommended[0].rules,
            'markdown/heading-increment': 'off', // allow headings to skip levels
            'markdown/fenced-code-language': 'off', // allow code blocks w/ no language specified
            'markdown/no-missing-label-refs': 'off' // allow missing label references
        }
    },
    { files: ['**/*.mjs'], languageOptions: { sourceType: 'module' }},
    { files: ['**/*.{yaml,yml}'], ...yml.configs['flat/standard'][1] }
]
```
</details>

```js
...
    speak(msg, { voice = 2, pitch = 2, speed = 1.1, onend } = {} ) {
    // Example call: chatgpt.speak(await chatgpt.getLastResponse(), { voice: 1, pitch: 2, speed: 3 })
    // - voice = index of voices available on user device
    // - pitch = float for pitch of speech from 0 to 2
    // - speed = float for rate of speech from 0.1 to 10
    // - onend = callback function when speech finishes

        // Validate args
        if (typeof msg != 'string') return console.error('Message must be a string!')
        const validOptionKeys = ['voice', 'pitch', 'speed', 'onend']
        for (const key in arguments[1]) {
            if (!validOptionKeys.includes(key))
                return console.error(`Invalid option '${key}'. Valid keys are: ${validOptionKeys}`)
            const val = arguments[1][key]
            if (key != 'onend' && typeof val != 'number' && !/^\d+$/.test(val))
                return console.error(`Invalid ${key} value '${val}'. Must be a number!`)
            else if (key == 'onend' && typeof val != 'function')
                return console.error(`Invalid ${key} value. Must be a function!`)
        }

        try { // to speak msg
            const utterance = new SpeechSynthesisUtterance()
            Object.assign(utterance, { text: msg, ...arguments[1], voice: speechSynthesis.getVoices()[voice] })
            speechSynthesis.speak(utterance)
        } catch (err) { console.error(err) }
    },
...
```

### What did you expect to happen?

No error

### What actually happened?

`no-unused-vars` error for 

```js
    speak(msg, { voice = 2, pitch = 2, speed = 1.1, onend } = {} ) {
```

![Image](https://github.com/user-attachments/assets/5b99f436-24de-4f85-92a3-55836c6a03c3)

...which are used here (e.g. `utterance.pitch` successfully uses `arguments[1].pitch`'s val):

```js
            Object.assign(utterance, { text: msg, ...arguments[1], voice: speechSynthesis.getVoices()[voice] })
```

### Link to Minimal Reproducible Example

https://github.com/KudoAI/chatgpt.js/actions/runs/13089810322/job/36525169442

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
This works as intended because variables `pitch`, `speed`, and `onend` are unused. Note that declaring these parameters and specifying default values has no effect on `arguments[1]`.

**Comment by adamlui** [comment]
What is the point of this error if it is working as intended? i.e. why do you feel it is wrong to specify default values, and what does it have anything to do with `no-unused-vars`

**Comment by adamlui** [comment]
Thanks this rule is truly useful!

**Comment by adamlui** [comment]
@mdjermanovic how come you responded so fast to both my issues this Saturday, also curious are you paid by ESLint?
